### DISCUSSION TITLE
Double devices not showing

### DISCUSSION THREAD

**Discussion by Roukie686868** [jziolkowski/tdm#179] [General]
I have 2 equal setups with both a ESP8266 and a 4 digit 7 segment displays. They both listen to the same full topic of course have different IP addresses as well they have a different Device name. Problem is that only one of them shows up in Tasmota Device Manager. It seems to be the unit with the lowest number in the 4th group of the IP address. Is there a way to show them both in Tasmota Device Manager even when they have the same full topic?

**Comment by barbudor** [comment]
Details of mqtt settings? Logs?
Crystal Ball is under maintenance...

**Reply by barbudor** [reply]
"Equal setup" may be your problem
They need to have different Topic and different MqttClient

**Reply by Roukie686868** [reply]
I run this on Windows and have no idea where to find a log file for this.
The two devices need to have the same Topic as they both have to do exactly the same. They have different Device Names as well have different Hostnames. But TDM0.2.11 lists only one of them at the time. I can see the line switch back and forth so on and then. I guess the last one sending an MQTT is the last one I see. Is there a way that TDM can show Devices based on both MQTT topic and DeviceName OR HostName?

**Reply by jziolkowski** [reply]
No, not really. To be usable with TDM, devices _must_ have unique topic. I know this may be limiting, but it is what it is.

**Reply by barbudor** [reply]
Tasmota devices Must have different topics.
Your problem is not with TDM but not following tasmota configuration recommandation

There are way to achieve what you are trying to do properly and without arm-twisting mqtt best practices.

Move this discussion to Tasmota's Github discussion

**Reply by Roukie686868** [reply]
Thanks for the info. I'll see what I can learn on MQTT best practices.

**Comment by Roukie686868** [comment]
Guys thanks for the feedback. Indeed I did set it up wrong. I now give them all individual topics and use a group topic to address them all at the same time.
